### PR TITLE
Fixes for getting the filename in the FileInfo class

### PR DESCRIPTION
**Pull request by hopleus**
## Summary
Fixes for getting the filename in the case when the file name key is received, but it is empty.

P.S. Faced with such a problem with ObjectStore enabled in S3 mode

## Checklist

- Code is [properly formatted](https://docs.nextcloud.com/server/latest/developer_manual/digging_deeper/continuous_integration.html#linting)
- [Sign-off message](https://github.com/src-d/guide/blob/master/developer-community/fix-DCO.md) is added to all commits
- [x] Tests ([unit](https://docs.nextcloud.com/server/latest/developer_manual/app_development/tutorial.html#unit-tests), [integration](https://docs.nextcloud.com/server/latest/developer_manual/app_development/tutorial.html#integration-tests), api and/or acceptance) are included
- [ ] Screenshots before/after for front-end changes
- [x] Documentation ([manuals](https://github.com/nextcloud/documentation/) or wiki) has been updated or is not required
- [x] [Backports requested](https://github.com/nextcloud/backportbot/#usage) where applicable (ex: critical bugfixes)

### CODE REVIEW

**Comment by hopleus** [comment]
@solracsf Hi. Could you run the tests again?

P.S. Last time cypress crashed with 503 errors and I didn't understand why

**Review by icewind1991 — APPROVED** [review]

File: lib/private/Files/FileInfo.php [hunk 1]
```diff
@@ -181,7 +181,11 @@ public function getMimePart() {
 	 * @return string
 	 */
 	public function getName() {
-		return isset($this->data['name']) ? $this->data['name'] : basename($this->getPath());
+		if (isset($this->data['name']) && !empty($this->data['name'])) {
+			return $this->data['name'];
+		}
+
+		return basename($this->getPath());
```
Comment (Altahrim) [inline lib/private/Files/FileInfo.php:188]: `empty` should be enough:
```suggestion
		return empty($this->data['name'])
			? basename($this->getPath())
			: $this->data['name'];
```

Comment (hopleus) [reply lib/private/Files/FileInfo.php]: > `empty` should be enough:

I agree, isset is redundant. I corrected the code

**Review by Altahrim — APPROVED** [review]

**Pull request merged**